src/Console/Output/ProcessCallback.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace PhpTuf\ComposerStager\Console\Output;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @internal
+ */
+final class ProcessCallback
+{
+    /**
+     * @var \Symfony\Component\Console\Output\OutputInterface
+     */
+    private $output;
+
+    /**
+     * @var \Symfony\Component\Console\Input\InputInterface
+     */
+    private $input;
+
+    public function __construct(InputInterface $input, OutputInterface $output)
+    {
+        $this->output = $output;
+        $this->input = $input;
+    }
+
+    /**
+     * @param string $type
+     *   The output type. Possible values are
+     *   \PhpTuf\ComposerStager\Domain\OutputType::ERR and
+     *   \PhpTuf\ComposerStager\Domain\OutputType::OUT.
+     * @param string $buffer
+     *   The message to output.
+     *
+     * @throws \Symfony\Component\Console\Exception\InvalidArgumentException
+     */
+    public function __invoke(string $type, string $buffer): void
```

Comment:
Do we ever use `$type`?

Reply:
Well, _we_ don't, per se. This is a Symfony Process callback: `function run(callable $callback = null, array $env = [])`. I've documented it as below wherever a callback is taken in the public API:

```php
/**
 * @param callable|null $callback An optional PHP callback to run whenever
 *   there is some output available on STDOUT or STDERR.
 *
 * @see https://symfony.com/doc/current/components/process.html#running-processes-asynchronously
 */
```

I'll add similar documentation here.